build.rs

```diff
@@ -39,15 +39,16 @@ fn main() {
 
     let now = Instant::now();
     let gen_path = Path::new(&out_dir).join("lib.rs");
-    rust_swig_expand(Path::new("src/lib.rs.in"), &gen_path);
+    let lib_rs_out = out_dir + "lib.rs.out";
+    expand_equality_macro(&lib_rs_out);
+    rust_swig_expand(Path::new(&lib_rs_out), &gen_path);
     let expand_time = now.elapsed();
     println!(
         "rust swig expand time: {}",
         expand_time.as_secs() as f64 + (expand_time.subsec_nanos() as f64) / 1_000_000_000.
     );
-    println!("cargo:rerun-if-changed=src");
-    //rebuild if user removes generated code
-    println!("cargo:rerun-if-changed={}", gen_path.display());
+    println!("cargo:rerun-if-changed=src/lib.rs.in");
```

Comment:
From what I was reading in the docs:

> Build scripts communicate with Cargo by printing to stdout. Cargo will interpret each line that starts with cargo: as an instruction that will influence compilation of the package.

Since this sort of output is hidden, you have to run `cargo b -vv` to see exactly what it's doing. 

I don't think we were using these commands well, so I made a small change to them. Now, it will only run the rust_swig expansion if one of three files is changed: lib.rs, lib.rs.in, or build.rs. If something else changed (like you're running the tests and want to make sure you ran `cargo b` with the latest changes), it won't take the ~30 seconds to re-build all the java files.